stock_picking_batch_creation/tests/test_clustering_conditions.py

```diff
@@ -0,0 +1,399 @@
+            Add a line to pick3
+            Create a new picking with 2 lines
+            Set the maximum number of preparation lines to 6 on the wizard.
+            Create a new batch where device3 will be used since the first pick
+            to prepare is pick3 and its volume is 30m3. -> device3 is the device
+        Expected result:
+            The batch should contain pick3, pick2 and pick1 but not the new picking
+            In this test case, only the number of lines is taken into account.
+            The sum of lines for pick3, pick1, pick1 is (3 + 1 + 1) = 5
+            The maximum number of preparation lines is 6.
+            The new picking is not added to the batch since its number of lines is 2.
+        """
+        self._set_quantity_in_stock(self.stock_location, self.p5)
+        self.p1.write(
+            {
+                "volume": 5.0,
+                "product_length": 5,
+                "product_height": 1,
+                "product_width": 1,
+                "weight": 1,
+            }
+        )
+        self.p2.write(
+            {
+                "volume": 5.0,
+                "product_length": 5,
+                "product_height": 1,
+                "product_width": 1,
+                "weight": 1,
+            }
+        )
+        self._add_product_to_picking(self.pick3, self.p5)
+        self.make_picking_batch.write({"maximum_number_of_preparation_lines": 6})
+
+        self._create_picking_pick_and_assign(
+            self.picking_type_1.id, products=self.p1 | self.p5
+        )
+        self.device3.nbr_bins = 60
+        batch = self.make_picking_batch._create_batch()
+        self.assertEqual(self.device3, batch.picking_device_id)
+        self.assertEqual(self.pick3 | self.pick2 | self.pick1, batch.picking_ids)
+
+    def test_no_device_for_clustering(self):
+        """
+        In this test we set the volume of all the products to 400m3.
+        Therefore, no device can be used to prepare the batch.
+        """
+        self.p1.write(
+            {
+                "product_length": 400,
+                "product_height": 1,
+                "product_width": 1,
+                "weight": 1,
+            }
+        )
+        self.p2.write(
+            {
+                "product_length": 400,
+                "product_height": 1,
+                "product_width": 1,
+                "weight": 1,
+            }
+        )
+        # since the volume is computed at picking creation or update, force
+        # the recomputation
+        (self.pick1 | self.pick2 | self.pick3).mapped("move_ids")._compute_volume()
+        batch = self.make_picking_batch._create_batch(raise_if_not_possible=False)
+        self.assertFalse(batch)
+        message = "No device found for batch picking."
+        with self.assertRaises(NoSuitableDeviceError) as cm:
+            self.make_picking_batch._create_batch(raise_if_not_possible=True)
+            self.assertEqual(cm.exception.name, message)
```

Comment:
```suggestion
            self.assertEqual(cm.exception, message)
```

As this provokes a warning